src/parser.rs

```diff
@@ -612,7 +613,50 @@ fn key_value_pairs(i: &[u8]) -> IResult<&[u8], HashMap<String, String>> {
     )(i)
 }
 
-fn key_value_pair(i: &[u8]) -> IResult<&[u8], (String, String)> {
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum QuotedOrUnquoted {
+    Unquoted(String),
+    Quoted(String),
+}
+
+impl QuotedOrUnquoted {
+    pub fn value(self) -> String {
```

Comment:
```suggestion
    pub fn into_string(self) -> String {
```

maybe, and also adding a `pub fn as_str(&self) -> &str` for good measure?

Reply:
eh, sorry. that doesn't even need to be there :)

Reply:
Well, it seems to be useful convenience API :)

Reply:
I'll think about... i mimicked what was implemented for `AlternativeMediaType` and it seemed sufficient.

Reply:
Can also be done in a separate PR but I think before releasing the next version we should make sure that all the attributes have useful types and those types have API that make them easy to deal with.